lib/src/facts/solaris/zone_resolver.cc

```diff
@@ -14,7 +14,10 @@ namespace facter { namespace facts { namespace solaris {
     zone_resolver::data zone_resolver::collect_data(collection& facts)
     {
         data result;
-        result.current_zone_name = get<1>(execute("/bin/zonename"));
```

Comment:
I think we previously always set `current_zone_name` even if the command failed, though it seems other resolvers check `success`, which makes sense, so I did the same. Not sure if we should be failing instead?